hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -49,7 +38,6 @@ func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1
 		NetworkInterfaces: []capiazure.NetworkInterface{{
 			SubnetName: subnetName,
 		}},
-		SSHPublicKey:  sshKey,
```

Comment:
Looks like updates to SSHPublicKey no longer triggers a rolling upgrade

Reply:
Ok it seems we use a different mechanism (i.e. MachineConfig) to propagate `hc.spec.sshKey` to the nodes, so AzureMachineTemplate.Template.Spec.SSHPublicKey is actually not used. This looks fine then